controllers/imagelist/imagelist_controller.go

```diff
@@ -86,19 +92,54 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return reconcile.Result{}, nil
 	}
 
-	imageList := &eraserv1alpha1.ImageList{}
-	err := r.Get(ctx, req.NamespacedName, imageList)
+	imageList := eraserv1alpha1.ImageList{}
+	err := r.Get(ctx, req.NamespacedName, &imageList)
 	if err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// If there is a change in ImageList, start ImageJob to trigger removal
+	jobList := eraserv1alpha1.ImageJobList{}
+	err = r.List(ctx, &jobList, &client.ListOptions{LabelSelector: labels.SelectorFromSet(
+		map[string]string{
+			consts.ImageJobOwnerLabelKey: imageList.Name,
+		},
+	)})
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+
+	switch len(jobList.Items) {
+	case 0:
+		return r.handleImageListEvent(ctx, &req, &imageList)
+	case 1:
+		return r.handleJobListEvent(ctx, &imageList, &jobList.Items[0])
```

Comment:
This needs additional logic. In addition to an ImageJob being moved to the `Completed` state, we will **also** reach this point when both of the following conditions are true:
1. Reconcile was triggered by an `UpdateEvent` to the `ImageList`
2. There is already and `ImageJob` in progress, and it is not yet complete.

This is related to #203 , and the best thing to do in that case is probably to requeue the `Imagelist`'s `UpdateEvent` for some reasonable time in the future, and to keep punting it until there are no `ImageJob`s in the running state.